src/game/menu.c

```diff
@@ -3413,20 +3413,37 @@ void menuClose(void)
 
 void func0f0f8120(void)
 {
+#ifdef AVOID_UB
+	u32 mpindex = g_MpPlayerNum % MAX_PLAYERS;
+	struct menudialog *prev = g_Menus[mpindex].curdialog;
```

Comment:
What is the undefined behavior here?

Reply:
When the player dies in single-player mode, `g_MpPlayerNum` is 4, whereas the array size is `MAX_PLAYERS` which is 4. There is at least one place elsewhere which detects this case and does appropriate correction for this case.

Reply:
Wouldn't the best fix be to have `mpindex` or `_MpPlayerNum` set to 0 globally? There are many other instances where the game can crash because `mpindex` or `_MpPlayerNum` is `5`.

Reply:
Arguably the "best" fix would be to rework the API so that `g_MpPlayerNum` isn't even used and an argument is passed to functions instead. But I don't think anything that drastic is being done at this stage.

There are cases where it's deliberately looking to see if it's 4 exactly. This is something that will need to be looked into as to *why* it's done that way. Who knows, maybe there's no need for the special case of 4.

I don't know where it's being set to 5.

Reply:
I had edited my comment to replace `5` with `4`. I will examine the code to see if there can't be a better fix.